### PR TITLE
Treatment effect component

### PR DESCRIPTION
**Pull request by collijk**
WARNING: not tested.

### CODE REVIEW

File: src/vivarium_csu_zenon/components/treatment/treatment.py [hunk 1]
```diff
@@ -209,3 +208,71 @@ def load_adherence_p(builder):
...
+
+    def on_initialize_simulants(self, pop_data: 'SimulantData'):
+        self.population_view.update(pd.Series(self.effect_size(pop_data.index),
+                                              index=pop_data.index,
+                                              name='initial_treatment_effect_size'))
+
+    def adjust_exposure(self, index, exposure):
+        initial_effect = self.population_view.subview(['initial_treatment_effect_size']).get(index)
+        initial_effect = initial_effect['initial_treatment_effect_size']  # coerce df to series
+        return exposure + initial_effect - self.effect_size(index)
```
Comment (aflaxman) [inline src/vivarium_csu_zenon/components/treatment/treatment.py]: LLT has multiplicative effects, so I expected something like `exposure * (1-initial_effect) * (1 - self.effect_size(index))`

https://vivarium-research.readthedocs.io/en/latest/gbd2017_models/concept_models/vivarium_sanofi_zenon/concept_model.html#initialization-strategy

> 
> Finally, we need to know: Untreated LDL-C level | treated, treatment profile t, adherence yes/no If not treated or not adherent, untreated LDL-c = LDC-c (drawn from GBD) If treated, and adherent, look up multiplier in treatment table 3 (second part): If monotherapy, treatment profile is {t}, eff_t ~ N(mean, sd) same for all individuals, and
> 
>     Untreated LDL-C = LDL-C / (1 - eff_t)
> 
> If FDC or multiple individual pills, treatment profile is {t, s}, eff_t ~ N, eff_s ~ N(mean_s, sd_s)
> 
>     Untreated LDL-C = LDL-C / ((1 - eff_t) * (1 - eff_s))
> 
> (same effect for all indiviuals, which we will note in the limitations)

File: src/vivarium_csu_zenon/components/treatment/treatment.py [hunk 2]
```diff
@@ -209,3 +208,71 @@ def load_adherence_p(builder):
...
+            parameters.HIGH_STATIN_LOW: pop_status[parameters.STATIN_HIGH] == 'low',
+            parameters.LOW_STATIN_HIGH: pop_status[parameters.STATIN_LOW] == 'high',
+            parameters.LOW_STATIN_LOW: pop_status[parameters.STATIN_LOW] == 'low',
+            parameters.EZETIMIBE: pop_status[parameters.EZETIMIBE],
+            parameters.FIBRATES: pop_status[parameters.FIBRATES],
+            parameters.LIFESTYLE: pop_status[parameters.LIFESTYLE],
+        }
+
+        for treatment, mask in treatment_profiles.items():
+            effect_size.loc[mask] += self.treatment_effect[treatment]
```
Comment (aflaxman) [inline src/vivarium_csu_zenon/components/treatment/treatment.py]: see comment above about LLT effect being multiplicative; based on that I was expecting something like `effect_size.loc[mask] *= (1-self.treatment_effect[treatment])`

File: src/vivarium_csu_zenon/components/treatment/treatment.py [hunk 3]
```diff
@@ -209,3 +208,71 @@ def load_adherence_p(builder):
...
+
+        return effect_size
+
+    @staticmethod
+    def load_treatment_effect(builder):
+        # FIXME: No data.
```
Comment (aflaxman) [inline src/vivarium_csu_zenon/components/treatment/treatment.py]: data is buried in one of the csvs the reseachers provided, which was supposed to appear in Table 3 https://vivarium-research.readthedocs.io/en/latest/gbd2017_models/concept_models/vivarium_sanofi_zenon/concept_model.html#id10  I will request an update to that!

**Comment by collijk** [comment]
All very helpful @aflaxman, thanks!

**Pull request merged**